enclave-runtime/src/lib.rs

```diff
@@ -468,12 +481,17 @@ where
 
 	let latest_onchain_header = validator.latest_finalized_header(validator.num_relays()).unwrap();
 
-	let rpc_author = Author::new(Arc::new(GlobalTopPoolContainer));
+	let author_mutex = GlobalAuthorContainer.get().ok_or_else(|| {
+		error!("Failed to retrieve author mutex. It might not be initialized?");
+		Error::MutexAccess
+	})?;
+
```

Comment:
The caller turns it into a `sgx_status_t` and loses the information about the mutex access afaict. That's why I decided to print it here.

Reply:
Yes, but the error is not lost. Although, it might not be detailed enough in some cases:

https://github.com/integritee-network/worker/blob/c60f460bf3ce2d019cb78e675ce19b0f6df1abc8/enclave-runtime/src/error.rs#L23-L31

Reply:
Here we give a warning about a conversion not happening as it should (which happens a lot in our case?), but it's not obvious that an error actually occurred?

Reply:
I think it is obvious that an error occurred. It will also transform the error into an sgx_error for top-level. So the ecall will also return an error.

Anyhow, I am fine with your approach, which is good also if you want to display more detailed msgs. I was just uncertain if you knew this existed. 👍

Reply:
Yes I did study what happens to the error, and I found it a bit weird that in most cases we will give a warning about the error not being able to be converted (and the error information), and then in the outer call we log the error, but without all the information. That's why I decided to log it as an error, explicitly.